pkg/dataloaders/dataloaders.go

```diff
@@ -0,0 +1,53 @@
+package dataloaders
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/vikstrous/dataloadgen"
+
+	"github.com/cmsgov/easi-app/pkg/authentication"
+	"github.com/cmsgov/easi-app/pkg/models"
+	"github.com/cmsgov/easi-app/pkg/storage"
+)
+
+type (
+	DataloaderFunc      func() *DataLoaders
+	fetchUserInfosFunc  func(ctx context.Context, euaUserIDs []string) ([]*models.UserInfo, error)
+	getCedarSystemsFunc func(ctx context.Context) ([]*models.CedarSystem, error)
+)
+
+type dataReader struct {
+	db              *storage.Store
+	fetchUserInfos  fetchUserInfosFunc
+	getCedarSystems getCedarSystemsFunc
+}
+
+type DataLoaders struct {
+	CedarSystemBookmark         *dataloadgen.Loader[models.BookmarkRequest, bool]
+	FetchUserInfos              *dataloadgen.Loader[string, *models.UserInfo]
+	GetUserAccounts             *dataloadgen.Loader[uuid.UUID, *authentication.UserAccount]
+	GetCedarSystems             *dataloadgen.Loader[string, *models.CedarSystem]
+	SystemIntakeContractNumbers *dataloadgen.Loader[uuid.UUID, []*models.SystemIntakeContractNumber]
+	SystemIntakeSystems         *dataloadgen.Loader[uuid.UUID, []*models.SystemIntakeSystem]
+	TRBRequestContractNumbers   *dataloadgen.Loader[uuid.UUID, []*models.TRBRequestContractNumber]
+	TRBRequestSystems           *dataloadgen.Loader[uuid.UUID, []*models.TRBRequestSystem]
+}
+
+func NewDataLoaders(store *storage.Store, fetchUserInfos fetchUserInfosFunc, getCedarSystems getCedarSystemsFunc) *DataLoaders {
```

Comment:
Can you add some documentation (comments) into this file covering the overall process for creating new dataloaders?

I think the most important part to cover is just the overall file/function structure -- i.e. that we have a `datareader` function that's not exported that handles the "batch" request, along with an exported function that handles the "single" (which is often called directly by the resolver)

Reply:
yeah! added! thanks Clay 😄